configurator.py

```diff
@@ -264,15 +264,21 @@ def get_full_code(self):
         return res
 
     def render_expression_record(self, expression, record):
-        template = Jinja2Template(expression, trim_blocks=True)
         try:
+            template = Jinja2Template(expression, trim_blocks=True)
             res = template.render(record)
         except TypeError as e:
             raise UserError(gettext(
                 'product_dynamic_configurator.msg_expression_error',
                 property=self.rec_name,
                 expression=expression[:25]+('...' if len(expression) > 25 else ''),
                 invalid=str(e)))
+        except TemplateSyntaxError as e:
```

Comment:
better reuse the previous "except" clause by using:

```
except (TypeError, TemplateSyntaxError) as e:
```

Reply:
40e08c11713070f65984f2efe7bf5e6401f000cc